test/test_conn.py

```diff
@@ -25,7 +32,7 @@ def _socket(mocker):
 
 
 @pytest.fixture
-def conn(_socket):
+def conn(_socket, dns_lookup):
```

Comment:
Took me a moment to realize this fixture _always_ provides a localhost... it'd be nice if there was a docstring explaining what this fixture is intended for and what it provides.

That's a more general comment about our fixtures though, as I've slowly waded through #1193 it's taken longer than it should have to follow the fixture paths because they aren't documented.